### PR TITLE
JAMES-2586 Implement Postgres upload repository

### PR DESCRIPTION
**Pull request by vttranlina**
resolve  https://github.com/linagora/james-project/issues/5015

### CODE REVIEW

**Comment by vttranlina** [comment]
TODO 

- [ ] Also add PostgresJmapUploadCleanupModule based on JmapUploadCleanupModule

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — COMMENTED** [review]
Please implement RLS based on the domain of the username?

File: server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/upload/CassandraUploadRepository.java [hunk 1]
```diff
@@ -91,10 +89,11 @@ public Flux<UploadMetaData> listUploads(Username user) {
             .map(UploadDAO.UploadRepresentation::toUploadMetaData);
     }
 
-    public Mono<Void> purge() {
-        Instant sevenDaysAgo = clock.instant().minus(EXPIRE_DURATION);
+    @Override
+    public Mono<Void> deleteByUploadDateBefore(Duration expireDuration) {
+        Instant expiredDayAgo = clock.instant().minus(expireDuration);
```
Comment (Arsnael) [inline server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/upload/CassandraUploadRepository.java:94]: `expiredDayAgo` => `expirationTime`?

**Review by hungphan227 — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Comment by vttranlina** [comment]
rebase master & squash fixup

**Comment by chibenwa** [comment]
Didn't had the time to review but I have no objection.

**Pull request merged**